src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonMetadataServices.cs

```diff
@@ -90,6 +84,15 @@ public static JsonPropertyInfo CreatePropertyInfo<T>(JsonSerializerOptions optio
         /// <remarks>This API is for use by the output of the System.Text.Json source generator and should not be called directly.</remarks>
         public static JsonTypeInfo<T> CreateValueInfo<T>(JsonSerializerOptions options, JsonConverter converter)
         {
+            if (options is null)
```

Comment:
Does this carry the potential of a breaking change? Is it possible that source generated code using .NET 6 tooling might pass a null argument in this method?

Reply:
we were previously throwing NRE here also note this is marked as non-nullable. I assume this API assumed we would throw somewhere else but we actually didn't